Sources/Opus/Opus.Decoder.swift

```diff
@@ -42,6 +42,9 @@ public extension Opus.Decoder {
 		try input.withUnsafeBytes {
 			let input = $0.bindMemory(to: UInt8.self)
 			let sampleCount = opus_decoder_get_nb_samples(decoder, input.baseAddress!, Int32($0.count))
+			if sampleCount < 0 {
+				throw Opus.Error.invalidPacket
```

Comment:
Could this be a new error with an associated value that's the bad sample count?

Reply:
It could if `Opus.Error` was an enum, in which case I could add a new case (for example: `invalidSampleCount`) with the sample count as the associated value. But, `Opus.Error` is a struct with static lets for the error types. Unless I'm missing something, I don't see how without rewriting the error type as an enum.
Unless you mean to do something similar to what you already do for decoding errors:
```Swift
if sampleCount < 0 {
    throw Opus.Error(sampleCount)
}
```

Reply:
Good point. Let’s keep it as you’ve written it.

Reply:
Looks like Opus clearly documented what happens if `opus_decoder_get_nb_samples` encounters a bad packet: https://www.opus-codec.org/docs/html_api/group__opusdecoder.html#ga895fbdd007bab726ace70be21f88f13e

> `OPUS_INVALID_PACKET` The compressed data passed is corrupted or of an unsupported type

So in this case, it should cast the return value to an `Opus.Error`. It’s small change, but will pass the precise error returned by the func.

Reply:
sounds good, I assume this is the change you mean:
```Swift
if sampleCount < 0 {
    throw Opus.Error(rawValue: sampleCount)
}
```

Reply:
Yeah. It *should* work without `rawValue:` (see elsewhere in this file).

Reply:
done!